napari/_vispy/vispy_base_layer.py

```diff
@@ -160,3 +163,18 @@ def _on_poll(self, event=None):
         VRAM or animating itself.
         """
         pass
+
+    def close(self):
+        """Vispy visual is closing."""
+        self.node.transforms = MatrixTransform()
+        self.node.parent = None
+        self.layer.events.refresh.disconnect(self._on_refresh_change)
+        self.layer.events.set_data.disconnect(self._on_data_change)
+        self.layer.events.visible.disconnect(self._on_visible_change)
+        self.layer.events.opacity.disconnect(self._on_opacity_change)
+        self.layer.events.blending.disconnect(self._on_blending_change)
+        self.layer.events.scale.disconnect(self._on_matrix_change)
+        self.layer.events.translate.disconnect(self._on_matrix_change)
+        self.layer.events.rotate.disconnect(self._on_matrix_change)
+        self.layer.events.shear.disconnect(self._on_matrix_change)
+        self.layer.events.affine.disconnect(self._on_matrix_change)
```

Comment:
Is this left over from the earlier version of this PR?

Reply:
changed